MapleServer2/Types/Guild/Guild.cs

```diff
@@ -0,0 +1,203 @@
+
+            Id = GuidGenerator.Long();
+            Name = name;
+            Capacity = (byte) property.Capacity;
+            Exp = 0;
+            Funds = 0;
+            Emblem = "";
+            Notice = "";
+            Searchable = true;
+            HouseRank = 1;
+            HouseTheme = 1;
+            Ranks = new GuildRank[6] {
+                new GuildRank("Master", 4095),
+                new GuildRank("Jr. Master"),
+                new GuildRank("Member 1"),
+                new GuildRank("Member 2"),
+                new GuildRank("New Member 1"),
+                new GuildRank("New Member 2") };
+            CreationTimestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds() + Environment.TickCount;
+
+            List<int> buffIds = GuildBuffMetadataStorage.GetBuffList();
+            foreach (int buffId in buffIds)
+            {
+                Buffs.Add(new GuildBuff(buffId));
+            }
+        }
+
+        public void AddLeader(Player player)
+        {
+            GuildMember leader = new GuildMember(player);
+
+            player.Guild = this;
+            leader.AddGuildMember(player);
+            Members.Add(leader);
+            player.GuildMember = leader;
+            leader.Rank = 0;
+            Leader = player;
+
+            DatabaseManager.Update(leader);
+            DatabaseManager.Update(this);
+            DatabaseManager.UpdateCharacter(player);
+        }
+
+        public void AddMember(Player player)
+        {
+            GuildMember member = new GuildMember(player);
+
+            member.Player = player;
+            player.Guild = this;
+            player.GuildMember = member;
+            Members.Add(member);
+
+            DatabaseManager.Update(member);
+            DatabaseManager.Update(this);
+            DatabaseManager.UpdateCharacter(player);
+        }
+
+        public void RemoveMember(Player player)
+        {
+            GuildMember member = Members.First(x => x.Player == player);
+            Members.Remove(member);
+
+            DatabaseManager.Delete(member);
+            DatabaseManager.Update(this);
+        }
+
+        public void AssignNewLeader(Player oldLeader, Player newLeader)
+        {
+            GuildMember newLeadMember = Members.First(x => x.Player == newLeader);
+            GuildMember oldLeadMember = Members.First(x => x.Player == oldLeader);
+
+            Members.Insert(0, newLeadMember);
+            Members.Remove(oldLeadMember);
+            Members.Add(oldLeadMember);
+            DatabaseManager.Update(this);
+        }
+
+        public void ModifyFunds(GameSession session, GuildPropertyMetadata property, int funds)
+        {
+            if (funds > 0)
+            {
+                if (Funds >= property.FundMax)
+                {
+                    return;
+                }
+                Funds += funds;
+
+                if (Funds >= property.FundMax)
+                {
+                    Funds = property.FundMax;
+                }
```

Comment:
Math.Min

Reply:
:thonk:

Reply:
Math.Min(x, y) gets the values which has the lowest value, exactly what you want here